en/03_Drawing_a_triangle/00_Setup/00_Base_code.adoc

```diff
@@ -51,51 +53,115 @@ int main() {
+a variety of standard exception types, as well, we catch the more general
+`std::exception`. One example of an error that we will deal with soon is finding
+out that a certain required extension is not supported.
+
+Roughly every chapter that follows after this one will add one new function that
+will be called from `initVulkan` and one or more new Vulkan objects to the
+private class members that need to be freed at the end in `cleanup`.
 
 == Resource management
 
-Just like each chunk of memory allocated with `malloc` requires a call to `free`, every Vulkan object that we create needs to be explicitly destroyed when we no longer need it.
-In C{pp} it is possible to perform automatic resource  management using https://en.wikipedia.org/wiki/Resource_Acquisition_Is_Initialization[RAII]  or smart pointers provided in the `<memory>` header.
-However, I've chosen to be explicit about allocation and deallocation of Vulkan objects in this tutorial.
-After all, Vulkan's niche is to be explicit about every operation to avoid mistakes, so it's good to be explicit about the lifetime of objects to learn how the API works.
+Just like each chunk of memory allocated with `malloc` requires a call to
+`free`, every Vulkan object that we create needs to be explicitly destroyed when
+we no longer need it. In c{pp} it is possible to perform automatic resource
+management using https://en.wikipedia.org/wiki/Resource_Acquisition_Is_Initialization[RAII]
+or smart pointers provided in the `<memory>` header. This tutorial is an attempt
+to make Vulkan easier to work with, and demonstrate modern Vulkan
+programming.  This tutorial will not only use RAII with smart pointers, it
+will endeavor to demonstrate the latest methods and extensions which should
+hopefully make Vulkan a joy to use.  Just because we enjoy working with
+low level graphics APIs, we shouldn't make the bar too high to learn how
+to do so.  Where appropriate, we will discuss concerns for resource
+management for freeing resources.  However, for this tutorial, we'll
+demonstrate that we can get pretty far with a basic destructor to clean up
+after our work.
+
+Vulkan objects are either created directly with functions like `vkCreateXXX`, or
+allocated through another object with functions like `vkAllocateXXX`. After
+making sure that an object is no longer used anywhere, you need to destroy it
+with the counterparts `vkDestroyXXX` and `vkFreeXXX`. The parameters for these
+functions generally vary for different types of objects, but there is one
+parameter that they all share: `pAllocator`. This is an optional parameter that
+allows you to specify callbacks for a custom memory allocator. We will ignore
+this parameter in the tutorial and always pass `nullptr` as argument.
+
+Using the Vulkan_hpp RAII module, we can rely upon the library to take care
+of `vkCreateXXX` `vkAllocateXXX` `vkDestroyXXX` and `vkFreeXXX` so a block
+of code that looks like this:
+
+[,c++]
+----
+        vkInstance instance;
```

Comment:
Minor nitpick: But I think we should remove indenation for standalone code blocks. Having snippets like these start every line with lots of empty space seems wasteful.